fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FeNameFormat.java

```diff
@@ -77,9 +98,21 @@ public static void checkPartitionName(String partitionName) throws AnalysisExcep
     }
 
     public static void checkColumnName(String columnName) {
-        if (Strings.isNullOrEmpty(columnName) || !columnName.matches(COLUMN_NAME_REGEX)) {
+        if (Strings.isNullOrEmpty(columnName)) {
             ErrorReport.reportSemanticException(ErrorCode.ERR_WRONG_COLUMN_NAME, columnName);
         }
+        String pattern;
+        if (RunMode.isSharedNothingMode()) {
```

Comment:
why column name patterns are different for shared nothing mode and shared data mode?

Reply:
For the sequence of predicates in the delete statement, Shared Nothing uses a string-based approach, which cannot distinguish special characters in column names. The PM considers this special process is reasonable .